machine/machine.go

```diff
@@ -191,6 +194,45 @@ func getNodeIdFromCpuBus(cpuBusPath string, threadId int) (int, error) {
 	return nodeId, nil
 }
 
+/* Look for per-node hugepages info using node id */
+/* Such as: /sys/devices/system/node/node%d/hugepages */
+func getHugePagesInfoFromNode(nodePath string, nodeIndex int) ([]info.HugePagesInfo, error) {
+	hugePagesInfo := []info.HugePagesInfo{}
+	path := filepath.Join(nodePath, fmt.Sprintf("node%d/hugepages", nodeIndex))
+	files, err := ioutil.ReadDir(path)
+	// Ignore if per-node info is not available.
+	if err != nil {
+		klog.Errorf("failed to get hugepages information for node %d: %v", nodeIndex, err)
+		return nil, nil
+	}
+
+	for _, file := range files {
```

Comment:
Is there a reason we cannot reuse some of the code from https://github.com/google/cadvisor/blob/f5e7ddf27add8bf6b3ead954d5b09d895a8597d7/machine/info.go#L66? The concept is the same here as it is there.

Reply:
can we consolidate how we parse capacity in `GetHugePagesInfo` in info.go with how we would parse here?  I am fine with the regex approach, I just want us to have a common approach.